python/cudf/cudf/_lib/pylibcudf/interop.pyx

```diff
@@ -170,6 +212,46 @@ def to_arrow(cudf_object, metadata=None):
     raise TypeError("to_arrow only accepts Table and Scalar objects")
 
 
+@to_arrow.register(DataType)
+def _to_arrow_datatype(cudf_object, **kwargs):
+    """
+    Convert a datatype to arrow.
+
+    Translation of some types requires extra information as a keyword
+    argument. Specifically:
+
+    - When translating a decimal type, provide ``precision``
+    - When translating a struct type, provide ``fields``
+    - When translating a list type, provide the wrapped ``value_type``
+    """
+    if cudf_object.id() in {type_id.DECIMAL32, type_id.DECIMAL64, type_id.DECIMAL128}:
+        if not (precision := kwargs.get("precision")):
+            raise ValueError(
+                "Precision must be provided for decimal types"
+            )
+            # no pa.decimal32 or pa.decimal64
+        return pa.decimal128(precision, -cudf_object.scale())
+    elif cudf_object.id() == type_id.STRUCT:
+        if not (fields := kwargs.get("fields")):
+            raise ValueError(
+                "Fields must be provided for struct types"
+            )
+        return pa.struct(fields)
+    elif cudf_object.id() == type_id.LIST:
+        if not (value_type := kwargs.get("value_type")):
+            raise ValueError(
+                "Value type must be provided for list types"
+            )
+        return pa.list_(value_type)
+    else:
+        try:
+            return ARROW_TO_PYLIBCUDF_TYPES[cudf_object.id()]
```

Comment:
Ah. I Tested everything except the normal scalar types 🥇

@wence- shall I put in a PR to fix this? seems like it would bloat https://github.com/rapidsai/cudf/pull/15890 a bit

Reply:
No, let's just do it as part of #15890 (I already have that...)